internal/controller/utils.go

```diff
@@ -77,6 +78,66 @@ func reconcileStatefulSet(ctx context.Context, logger logr.Logger, ec *ecv1alpha
 	return getStatefulSet(ctx, c, ec.Name, ec.Namespace)
 }
 
+func setDefaultArgs(name string) []string {
+	return []string{
+		"--name=$(POD_NAME)",
+		"--listen-peer-urls=http://0.0.0.0:2380",   // TODO: only listen on 127.0.0.1 and host IP
+		"--listen-client-urls=http://0.0.0.0:2379", // TODO: only listen on 127.0.0.1 and host IP
+		fmt.Sprintf("--initial-advertise-peer-urls=http://$(POD_NAME).%s.$(POD_NAMESPACE).svc.cluster.local:2380", name),
+		fmt.Sprintf("--advertise-client-urls=http://$(POD_NAME).%s.$(POD_NAMESPACE).svc.cluster.local:2379", name),
+	}
+}
+
+func RemoveStringFromSlice(defaultArgs []string, s string) []string {
```

Comment:
```suggestion
func RemoveStringFromSlice(defaultArgs []string, customArg string) []string {
```

Reply:
I've been  thinking about this and think it better to make the function more generic. Changing line 91 to
`func RemoveStringFromSlice(s []string, str string) []string` would make the inputs better for use cases outside of custom args.